geode-core/src/distributedTest/java/org/apache/geode/internal/tcp/ConnectionDUnitTest.java

```diff
@@ -20,35 +20,65 @@
 
 import java.io.ByteArrayInputStream;
 import java.io.DataInputStream;
+import java.io.File;
 import java.net.InetSocketAddress;
 import java.net.Socket;
 import java.nio.ByteBuffer;
+import java.util.Properties;
 
-import org.junit.Assert;
-import org.junit.Ignore;
+import org.junit.After;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.contrib.java.lang.system.SystemOutRule;
 import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
 import org.mockito.junit.MockitoJUnit;
 import org.mockito.junit.MockitoRule;
 
+import org.apache.geode.distributed.ConfigurationProperties;
+import org.apache.geode.distributed.DistributedSystem;
 import org.apache.geode.distributed.internal.DistributionConfig;
 import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
 import org.apache.geode.internal.net.SocketCloser;
+import org.apache.geode.test.assertj.LogFileAssert;
+import org.apache.geode.test.dunit.IgnoredException;
+import org.apache.geode.test.dunit.internal.DUnitLauncher;
+import org.apache.geode.test.dunit.rules.DistributedRule;
 import org.apache.geode.test.junit.categories.MembershipTest;
 
 @Category({MembershipTest.class})
-public class ConnectionTest {
+public class ConnectionDUnitTest {
+
+  @Rule
+  public final DistributedRule distributedRule = DistributedRule.builder().withVMCount(0).build();
+
   @Rule
-  public MockitoRule mockitoRule = MockitoJUnit.rule().strictness(STRICT_STUBS);
+  public final MockitoRule mockitoRule = MockitoJUnit.rule().strictness(STRICT_STUBS);
 
   @Rule
-  public SystemOutRule systemOutRule = new SystemOutRule().enableLog();
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private File logFile;
+
+  private DistributedSystem distributedSystem;
+
+  @After
+  public void teardown() {
+    if (distributedSystem != null) {
+      distributedSystem.disconnect();
+    }
+  }
 
   @Test
-  @Ignore("Test ignored: SystemOutRule fails to capture output logged by Connection in CI runs")
-  public void badHeaderMessageIsCorrectlyLogged() {
+  public void badHeaderMessageIsCorrectlyLogged() throws Exception {
+    Properties properties = DUnitLauncher.getDistributedSystemProperties();
+    logFile = temporaryFolder.newFile();
+    properties.put(ConfigurationProperties.LOG_FILE, logFile.getAbsolutePath());
+    distributedSystem = DistributedSystem.connect(properties);
```

Comment:
The `connect` method is deprecated, I'd prefer to avoid adding new code that uses deprecated stuff.

Reply:
Yeah, okay.  Personally I think that shouldn't be deprecated.  We could build other things besides a Cache on a DistributedSystem and we actually started doing that at one point.